src/lib/storage/index/abstract_index.cpp

```diff
@@ -0,0 +1,77 @@
+#include "abstract_index.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/index/b_tree/b_tree_index.hpp"
+#include "storage/index/group_key/composite_group_key_index.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+
+namespace opossum {
+
+size_t AbstractIndex::estimate_memory_consumption(SegmentIndexType type, ChunkOffset row_count,
+                                                  ChunkOffset distinct_count, uint32_t value_bytes) {
+  switch (type) {
+    case SegmentIndexType::GroupKey:
+      return GroupKeyIndex::estimate_memory_consumption(row_count, distinct_count, value_bytes);
+    case SegmentIndexType::CompositeGroupKey:
+      return CompositeGroupKeyIndex::estimate_memory_consumption(row_count, distinct_count, value_bytes);
+    case SegmentIndexType::AdaptiveRadixTree:
+      return AdaptiveRadixTreeIndex::estimate_memory_consumption(row_count, distinct_count, value_bytes);
+    case SegmentIndexType::BTree:
+      return BTreeIndex::estimate_memory_consumption(row_count, distinct_count, value_bytes);
+    default:
```

Comment:
Is this the case for any index? If not, remove it the default. It's usually better to fail at compile-time.